Update admin views and voting new

s-espinosa opened the pull request:
@notmarkmiranda Any idea what's going on with the view for a vote page? I literally can't see the radio buttons.

notmarkmiranda commented:
@s-espinosa i dont know? i've never actually looked at it. is it an admin page? i'm sure it's related to materialize, but the tests pass?

notmarkmiranda commented:
@s-espinosa sidenote: why are you using `select_tag` instead of `f.select`?

s-espinosa commented:
I was using `select_tag` instead of `f.select` because the ratings aren't choices from a model (e.g. they're not separate rows in a `Ratings` table), but just number options that someone can give a project. That's what I consider the main difference between the two. Can you provide a code snippet that would work with `f.select`?

s-espinosa commented:
@notmarkmiranda Believe this is ready for review. Let me know what you think.

# Primary Changes
## Admin Demo Nights
* Controller
    * Add `update`
    * Switch leftover `active` reference to `status`
* Index
    * Make links to change status functional.
    * Create a section to show closed demo-nights
* Show
    * Add buttons to change status of demo night
    * Replaced text names of projects associated with the DemoNight to links

## Admin Projects
* All New
    * Add `show`, `edit`, and `update` views/controller actions.
* Index
    * Update to only show current projects

## Models
* Projects
    * Add method to select only projects associated with the current DemoNight

# Secondary Changes
* Added a `sleep(0.1)` to one of the tests that seemed to not be seeing dropdowns when run as part of the suite. Seems ridiculous, but let me know if you have a better solution/see the same issue without it.
* Added more robust descriptions to voting categories

notmarkmiranda commented on app/views/admin/demo_nights/index.html.erb:
```diff
@@ -1,15 +1,23 @@
+    <%= "#{dn.name} | #{dn.status.humanize}" %>
     <% if dn.status != "accepting_submissions" %>
-      <%= link_to "switch to accepting submissions", "#", class: "waves-effect waves-light btn turing-btn" %>
+      <%= link_to "accept submissions", admin_demo_night_path(dn, demo_night: {status: :accepting_submissions}), method: :patch, class: "waves-effect waves-light btn turing-btn" %>
```
I don't know how I feel about this. It could be moved, but would only make the view better and the controller worse?

notmarkmiranda commented on app/views/votes/new.html.erb:
```diff
@@ -1,19 +1,18 @@
       <div class="input-field col s12">
         <%= f.select(:representation, [1, 2, 3, 4, 5], selected: 3) %>
-        <%= f.label :representation %>
+        <%= f.label "Does this project represent the student community?" %>
```
I'm not sure if this will make a difference, but I think this should be: 
`f.label :representation, "Does this project represent the student community?"`

notmarkmiranda commented on spec/features/admin_can_create_demo_night_spec.rb:
```diff
@@ -17,7 +17,7 @@
 
     within('.admin-tools') do
-      expect(page).to have_link('Activate Demo Night!')
+      expect(page).to have_link('Open voting!')
```
Good use of a `!`

notmarkmiranda commented:
made a couple comments, nothing huge. I think it looks good.
